### DISCUSSION TITLE
Not able to set the configuration

### DISCUSSION THREAD

**Discussion by gaurang-chandgothia** [libusb/libusb#1424] [Q&A]
I have a USB Device which i wish to access through libusb. I have made a project in QT 5.10 (QT Creator 4.5) in which my main file contains three functions as APP_INIT, APP_DEINIT and main. Under the function APP_INIT, i am initializing libusb, opening the device, reading its descripters, setting the configuration and claimimng the interface. libusb is getting initialized, device is opened with the specified PID/VID and descriptors are read successfully. However, when i am trying to set the configuration, the function libusb_set_configuration return -5 representing LIBUSB_ERROR_NOT_FOUND (entity not found). The code, header file and QT Project FIle (.pro) is attached for ready reference.
[Libusb_code.txt](https://github.com/libusb/libusb/files/13803796/Libusb_code.txt)
[qt_project_pro_file.txt](https://github.com/libusb/libusb/files/13803797/qt_project_pro_file.txt)
[hostlib.txt](https://github.com/libusb/libusb/files/13803807/hostlib.txt)

I am using libusb 1.0.22

I also created a project in visual studio, but encountering the same error there also.

Please help out in identifying the problem.

**Comment by Youw** [comment]
What makes you think you should be able to set the the configuration at all? Does it exist in the descriptor?

NOTE: this is a libusb _issue tracker_, not a support forum.
At most you should use the "Discussions" section, but more preferable is to use libusb mailing list/StackOverflow/other forums to seek a device-specific help.
